### DISCUSSION TITLE
Using SSO token to auth to internal endpoints

### DISCUSSION THREAD

**Discussion by cbutler-captech** [danny-avila/LibreChat#2665] [Help Wanted]
We have an API Management instance that is set up to authorize via the Azure OpenID token. This API has the keys that are then added and model security and quota are applied. What would be the best strategy to implement this inside the codebase.  I was browsing the code and considered doing this in the BaseClient but wanted to check here and see how a developer would attack this.

**Comment by danny-avila** [comment]
You can also try using express.js middleware, you can apply it to many routes at once like this.
